rust/agama-cli/src/lib.rs

```diff
@@ -185,15 +187,24 @@ async fn build_manager<'a>() -> anyhow::Result<ManagerClient<'a>> {
     Ok(ManagerClient::new(conn).await?)
 }
 
-pub fn download_file(url: &str, path: &PathBuf) -> Result<(), ServiceError> {
+pub fn download_file(url: &str, path: &PathBuf) -> anyhow::Result<()> {
     let mut file = fs::OpenOptions::new()
         .create(true)
         .write(true)
-        .mode(0o400)
+        .truncate(true)
+        .mode(0o600)
         .open(path)
         .context(format!("Cannot write the file '{}'", path.display()))?;
 
-    match Transfer::get(url, &mut file) {
+    let context = InstallationContext::from_env().unwrap();
+    let uri = UriRef::parse(url).context("Invalid URL")?;
+    let absolute_url = if uri.has_scheme() {
+        uri.to_string()
+    } else {
+        uri.resolve_against(&context.source)?.to_string()
```

Comment:
Documentation! Again we are adding a secret feature. Please locate the 2 occurrences of `relurl` in Agama source, both are affected by this PR, one by this commit :)